File: backend/api/models/company.py
"""
Company-related Pydantic models.
"""

from typing import Optional

from pydantic import BaseModel, Field


class Company(BaseModel):
    """Company metadata model."""
    id: str = Field(..., description="Unique company identifier")
    name: str = Field(..., description="Full company name")
    ticker: str = Field(..., description="Stock ticker symbol")
    sector: str = Field(..., description="Industry sector")
    market_cap: str = Field(..., description="Market capitalization (formatted)")
    logo: Optional[str] = Field(None, description="Logo URL or identifier")
    description: Optional[str] = Field(None, description="Company description")

File: frontend/src/types/company.ts
export type Company = {
  id: string;
  name: string;
  ticker: string;
  sector: string;
  marketCap: string;
  logo?: string;
  historicalNewsAvailable?: boolean; // FNSPID data (2000-2023)
  etfSubSector?: string; // MOAT ETF technology sub-sector
  isMoatEtfHolding?: boolean;
};

